### PR TITLE
Parse rewrite

### PR DESCRIPTION
**Pull request by dreulavelle**
- [Move parser to its own module](https://github.com/dreulavelle/iceberg/commit/30ecdc31d730d0384e9b9a00c9fe08950c997615)

- [Fix overseerr, watchlist, jackett validation.](https://github.com/dreulavelle/iceberg/commit/0ea39febf4d8153543daf8ec2bc833d520541229)

- [Added more refined logic to parser module.](https://github.com/dreulavelle/iceberg/commit/ec7f55096fd636a03b320079327be0665e9b87e8)

- [Set stage for testing](https://github.com/dreulavelle/iceberg/commit/c57e6a49e8fd24a8a6c5fdf9a580eab5abc12466)

- [Add methods for individual checks](https://github.com/dreulavelle/iceberg/commit/f1fc80ad70d606964cbb371e3fb47510dc525216)

- [Update sort logic](https://github.com/dreulavelle/iceberg/commit/6127d3dbb62688f6b3c57495912c76c7365486d6)

- [Update default settings](https://github.com/dreulavelle/iceberg/commit/1671dd5a38ec3758d9f3e43ed0a4ad233985f9d5)

Needs to be tested further. Works good so far from what I can see. It's removing items appropriately as well as sorting items as well. There can be a lot more done with it, but its a good start for now.

Logic has changed from parsing per title, to parsing per title + sorting the dict after all the items have been scraped by a scraper.

### CODE REVIEW

**Comment by dreulavelle** [comment]
Dealt with merge conflicts.

File: backend/program/scrapers/__init__.py [hunk 1]
```diff
@@ -55,3 +56,13 @@ def _needs_new_scrape(self, item) -> bool:
             > scrape_time
             or item.scraped_times == 0
         )
+    def _check_for_title_match(self, item, string) -> bool:
```
Comment (Gaisberg) [inline backend/program/scrapers/__init__.py:59]: whats this, its not being used?

File: backend/program/scrapers/jackett.py [hunk 2]
```diff
@@ -35,7 +35,7 @@ def validate_settings(self) -> bool:
             try:
                 url = f"{self.settings.url}/api/v2.0/server/config"
                 response = get(url=url, retry_if_failed=False, timeout=60)
-                if response.is_ok:
+                if response.is_ok and response.data.api_key is not None:
```
Comment (Gaisberg) [inline backend/program/scrapers/jackett.py:38]: None check first

**Review by Gaisberg — CHANGES_REQUESTED** [review]

File: backend/program/scrapers/jackett.py (hunk 2 above)
Comment (dreulavelle) [reply backend/program/scrapers/jackett.py:38]: oh swap ordering?

File: backend/program/scrapers/__init__.py (hunk 1 above)
Comment (dreulavelle) [reply backend/program/scrapers/__init__.py:59]: It's going to be, just getting it ready to switch jackett/prowlarr to keyword queries

File: backend/program/scrapers/jackett.py (hunk 2 above)
Comment (Gaisberg) [reply backend/program/scrapers/jackett.py:38]: Oh, sorry nvm...

**Review by Gaisberg — APPROVED** [review]

**Pull request merged**